doc/changelog.md

```diff
@@ -1,3 +1,55 @@
+# Release 0.4.1
+
+<h3>Improvements</h3>
+
+* Catalyst wheels are now packaged with OpenMP and ZStd.
+  [(#457)](https://github.com/PennyLaneAI/catalyst/pull/457)
+
+<h3>Bug fixes</h3>
+
+* Resolve an infinite recursion in the decomposition of the `Controlled`
+  operator whenever computing a Unitary matrix for the operator fails.
+  [(#468)](https://github.com/PennyLaneAI/catalyst/pull/468)
+
+* Resolve a failure to generate gradient code for specific input circuits.
+  [(#439)](https://github.com/PennyLaneAI/catalyst/issues/439)
+
+  In this case, [`jnp.mod`](https://github.com/PennyLaneAI/catalyst/issues/437)
+  was used to compute wire values in a for loop, which prevented the gradient
+  architecture from fully separating quantum and classical code. The following
+  program is now supported:
+  ```py
+  @qjit
+  @grad
+  @qml.qnode(dev)
+  def f(x):
+      def cnot_loop(j):
+          qml.CNOT(wires=[j, jnp.mod((j + 1), 4)])
+  
+      for_loop(0, 4, 1)(cnot_loop)()
+  
+      return qml.expval(qml.PauliZ(0))
+  ```
+
+* Resolve unpredictable behaviour when importing libraries that share Catalyst's LLVM dependency
+  (e.g. TensorFlow). In some cases, both packages exporting the same symbols from their shared
+  libraries can lead to process crashes and other unpredictable behaviour, since the wrong functions
+  can be called if both libraries are loaded in the current process.
+  The fix involves building shared libraries with hidden (macOS) or protected (linux) symbol
+  visibility by default, exporting only what is necessary.
+  [(#465)](https://github.com/PennyLaneAI/catalyst/pull/465)
+
+* Resolve a failure to find the SciPy OpenBLAS library when running Catalyst,
+  due to a different SciPy version being used to build Catalyst than to run it.
+  [(#471)](https://github.com/PennyLaneAI/catalyst/pull/471)
+
```

Comment:
```suggestion

* Fix the issue in `LightningKokkos::AllocateQubits` with allocating too many qubit IDs on
  qubit re-allocation.
  [(#473)](https://github.com/PennyLaneAI/catalyst/pull/473)

```

Reply:
@maliasadi I don't believe I included this PR either

Reply:
yup, you are right! will you also include that PR?

Reply:
Nope, I want to avoid pulling in too many PRs :)